pennylane/transforms/qcut.py

```diff
@@ -364,6 +365,174 @@ def graph_to_tape(graph: MultiDiGraph) -> QuantumTape:
+        )
+    if n_measurements == 0:
+        return [expval(g) for g in group]
+
+    measurement = measurements[0]
+
+    if measurement.return_type is not Expectation:
+        raise ValueError(
+            "The circuit cutting workflow only supports circuits with expectation "
+            "value measurements"
+        )
+
+    obs = measurement.obs
+
+    return [expval(obs @ g) for g in group]
+
+
+def _prep_zero_state(wire):
+    Identity(wire)
+
+
+def _prep_one_state(wire):
+    PauliX(wire)
+
+
+def _prep_plus_state(wire):
+    Hadamard(wire)
+
+
+def _prep_iplus_state(wire):
+    Hadamard(wire)
+    S(wires=wire)
+
+
+PREPARE_SETTINGS = [_prep_zero_state, _prep_one_state, _prep_plus_state, _prep_iplus_state]
+
+
+def expand_fragment_tapes(
+    tape: QuantumTape,
+) -> Tuple[List[QuantumTape], List[PrepareNode], List[MeasureNode]]:
+    """
+    Expands a fragment tape into a collection of tapes for each configuration of
+    :class:`MeasureNode` and :class:`PrepareNode` operations.
+
+    .. note::
+
+        This function is designed for use as part of the circuit cutting workflow. Check out the
+        :doc:`transforms </code/qml_transforms>` page for more details.
+
+    Args:
+        tape (QuantumTape): the fragment tape to be expanded.
+
+    Returns:
+        Tuple[List[QuantumTape], List[PrepareNode], List[MeasureNode]]: the
+        tapes corresponding to each configuration, the preparation nodes and
+        the measurement nodes.
+
+    **Example**
+
+    Consider the following circuit, which contains a :class:`~.MeasureNode` and :class:`~.PrepareNode`
+    operation:
+
+    .. code-block:: python
+
+        from pennylane.transforms import qcut
+
+        with qml.tape.QuantumTape() as tape:
+            qcut.PrepareNode(wires=0)
+            qml.RX(0.5, wires=0)
+            qcut.MeasureNode(wires=0)
+
+    We can expand over the measurement and preparation nodes using:
+
+    .. code-block:: python
+
+        >>> tapes, prep, meas = qml.transforms.expand_fragment_tapes(tape)
+        >>> for t in tapes:
+        ...     print(t.draw())
+         0: ──I──RX(0.5)──┤ ⟨I⟩ ┤ ⟨Z⟩
+
+         0: ──I──RX(0.5)──┤ ⟨X⟩
+
+         0: ──I──RX(0.5)──┤ ⟨Y⟩
+
+         0: ──X──RX(0.5)──┤ ⟨I⟩ ┤ ⟨Z⟩
+
+         0: ──X──RX(0.5)──┤ ⟨X⟩
+
+         0: ──X──RX(0.5)──┤ ⟨Y⟩
+
+         0: ──H──RX(0.5)──┤ ⟨I⟩ ┤ ⟨Z⟩
+
+         0: ──H──RX(0.5)──┤ ⟨X⟩
+
+         0: ──H──RX(0.5)──┤ ⟨Y⟩
+
+         0: ──H──S──RX(0.5)──┤ ⟨I⟩ ┤ ⟨Z⟩
+
+         0: ──H──S──RX(0.5)──┤ ⟨X⟩
+
+         0: ──H──S──RX(0.5)──┤ ⟨Y⟩
+    """
+    prepare_nodes = [o for o in tape.operations if isinstance(o, PrepareNode)]
+    measure_nodes = [o for o in tape.operations if isinstance(o, MeasureNode)]
+
+    wire_map = {mn.wires[0]: i for i, mn in enumerate(measure_nodes)}
+
+    n_meas = len(measure_nodes)
+    if n_meas >= 1:
+        measure_combinations = qml.grouping.partition_pauli_group(len(measure_nodes))
+    else:
+        measure_combinations = [[""]]
```

Comment:
We should update this when we merge in https://github.com/PennyLaneAI/pennylane/pull/2192